packages/react-router-dom/__tests__/nav-link-active-test.tsx

```diff
@@ -288,6 +288,57 @@ describe("NavLink", () => {
 
       expect(anchors.map((a) => a.props.className)).toEqual(["active", ""]);
     });
+
+    it("does not automatically apply to root non-layout segments", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home"]}>
+            <Routes>
+              <Route index element={<h1>Root</h1>} />
+              <Route
+                path="home"
+                element={<NavLink to="/">Root</NavLink>}
+              ></Route>
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      let anchor = renderer.root.findByType("a");
+
+      expect(anchor.props.className).not.toMatch("active");
+    });
+
+    it("does not automatically apply to root layout segments", () => {
+      let renderer: TestRenderer.ReactTestRenderer;
+      TestRenderer.act(() => {
+        renderer = TestRenderer.create(
+          <MemoryRouter initialEntries={["/home"]}>
+            <Routes>
+              <Route
+                path="/"
+                element={
+                  <>
+                    <h1>Root</h1>
+                    <Outlet />
+                  </>
+                }
+              >
+                <Route
+                  path="home"
+                  element={<NavLink to="/">Root</NavLink>}
+                ></Route>
+              </Route>
+            </Routes>
+          </MemoryRouter>
+        );
+      });
+
+      let anchor = renderer.root.findByType("a");
+
+      expect(anchor.props.className).not.toMatch("active");
```

Comment:
@mjackson Wanted to get your thoughts on the behavior of `<NavLink to="/">`.  In 6.3 and before it seems like it was impossible to get that to apply as a partial match (`end=false`).  Assuming that's intentional (since `/` is a weird case and matches _everything_), then this change reverts back to the previous logic from 6.3 and updates it for `isPending`.

Reply:
This looks right to me. A `<NavLink to="/">` should *not* be active at `/home`.